Sources/Nimble/Swift3to22.swift

```diff
@@ -0,0 +1,157 @@
+import Foundation
+import XCTest
+
+#if !swift(>=3)
+    public typealias Boolean = BooleanType
+    public typealias Collection = CollectionType
+    public typealias ErrorProtocol = ErrorType
+
+    internal func unsafeBitCast<T, U>(x: T, to: U.Type) -> U {
+        return unsafeBitCast(x, to)
+    }
+
+    typealias NSFastEnumerationIterator = NSFastGenerator
+
+    extension NSObjectProtocol {
+        internal func isKind(of aClass: Swift.AnyClass) -> Bool {
+            return isKindOfClass(aClass)
+        }
+
+        internal func isMember(of aClass: Swift.AnyClass) -> Bool {
+            return isMemberOfClass(aClass)
+        }
+    }
+
+    public protocol Sequence: SequenceType {
```

Comment:
For supporting uses `associatedtype` constraints on some functions, `Sequence` is defined as `protocol` inheriting `SequenceType`.
That would require some `SequenceType` types defined on consumer to conform as:
```swift
#if !swift(>=3)
extension MyCustomSequence: Nimble.Sequence {}
#endif
```